src/components/structures/InteractiveAuth.js

```diff
@@ -177,7 +177,13 @@ export default class InteractiveAuthComponent extends React.Component {
             stageState: stageState,
             errorText: stageState.error,
         }, () => {
-            if (oldStage != stageType) this._setFocus();
+            if (oldStage !== stageType) {
+                this._setFocus();
+            } else if (!stageState.error && this._stageComponent.current &&
+                this._stageComponent.current.attemptFailed
+            ) {
```

Comment:
Hmm, I find these multi-line conditionals a bit more readable when all lines start at the same column:

```suggestion
            } else if (
                !stageState.error && this._stageComponent.current &&
                this._stageComponent.current.attemptFailed
            ) {
```